### PR TITLE
feat(components): [tree-v2] add `props.class` prop

### LINKED ISSUE #16780: [Style] [tree-v2] V2 Tree node custom class
Bug Type: **`Style`**

## Environment

- Vue Version: `3.5.0-alpha.2`
- Element Plus Version: `2.7.2`
- Browser / OS: `Chrome/124.0.0.0`
- Build Tool: `Vite`

### Related Component

- `el-tree-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cImN1c3RvbS10cmVlLW5vZGUtY29udGFpbmVyXCI+XG4gIDxzcGFuPlRSRUU8L3NwYW4+XG4gICAgPGVsLXRyZWVcbiAgICAgIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiXG4gICAgICA6ZGF0YT1cImRhdGFcIlxuICAgICAgc2hvdy1jaGVja2JveFxuICAgICAgbm9kZS1rZXk9XCJpZFwiXG4gICAgICBkZWZhdWx0LWV4cGFuZC1hbGxcbiAgICAgIDpleHBhbmQtb24tY2xpY2stbm9kZT1cImZhbHNlXCJcbiAgICAgIDpwcm9wcz1cInsgY2xhc3M6IGN1c3RvbU5vZGVDbGFzcyB9XCJcbiAgICAvPlxuICA8c3Bhbj5UUkVFIFYyPC9zcGFuPlxuXG4gICAgICA8ZWwtdHJlZS12MlxuICAgIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiXG4gICAgOmRhdGE9XCJkYXRhdjJcIlxuICAgIDpwcm9wcz1cInByb3BzdjJcIlxuICAgIDpoZWlnaHQ9XCIyMDhcIlxuICAvPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgdHlwZSBOb2RlIGZyb20gJ2VsZW1lbnQtcGx1cy9lcy9jb21wb25lbnRzL3RyZWUvc3JjL21vZGVsL25vZGUnXG5cbmludGVyZmFjZSBUcmVlIHtcbiAgaWQ6IG51bWJlclxuICBsYWJlbDogc3RyaW5nXG4gIGlzUGVudWx0aW1hdGU/OiBib29sZWFuXG4gIGNoaWxkcmVuPzogVHJlZVtdXG59XG5cbmNvbnN0IGN1c3RvbU5vZGVDbGFzcyA9IChkYXRhOiBUcmVlLCBub2RlOiBOb2RlKSA9PiB7XG4gIGlmIChkYXRhLmlzUGVudWx0aW1hdGUpIHtcbiAgICByZXR1cm4gJ2lzLXBlbnVsdGltYXRlJ1xuICB9XG4gIHJldHVybiBudWxsXG59XG5cbmNvbnN0IGRhdGE6IFRyZWVbXSA9IFtcbiAge1xuICAgIGlkOiAxLFxuICAgIGxhYmVsOiAnTGV2ZWwgb25lIDEnLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIGlkOiA0LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAxLTEnLFxuICAgICAgICBpc1BlbnVsdGltYXRlOiB0cnVlLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGlkOiA5LFxuICAgICAgICAgICAgbGFiZWw6ICdMZXZlbCB0aHJlZSAxLTEtMScsXG4gICAgICAgICAgfSxcbiAgICAgICAgICB7XG4gICAgICAgICAgICBpZDogMTAsXG4gICAgICAgICAgICBsYWJlbDogJ0xldmVsIHRocmVlIDEtMS0yJyxcbiAgICAgICAgICB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgaWQ6IDIsXG4gICAgbGFiZWw6ICdMZXZlbCBvbmUgMicsXG4gICAgaXNQZW51bHRpbWF0ZTogdHJ1ZSxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogNSxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMi0xJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiA2LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAyLTInLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgaWQ6IDMsXG4gICAgbGFiZWw6ICdMZXZlbCBvbmUgMycsXG4gICAgaXNQZW51bHRpbWF0ZTogdHJ1ZSxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogNyxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMy0xJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiA4LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAzLTInLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuXG5cblxuLy8vLy8vLy8vLy8vIHYyIHRyZWVcblxuaW50ZXJmYWNlIFRyZWVWMiB7XG4gIGlkOiBzdHJpbmdcbiAgbGFiZWw6IHN0cmluZ1xuICBjaGlsZHJlbj86IFRyZWVbXVxufVxuXG5jb25zdCBnZXRLZXkgPSAocHJlZml4OiBzdHJpbmcsIGlkOiBudW1iZXIpID0+IHtcbiAgcmV0dXJuIGAke3ByZWZpeH0tJHtpZH1gXG59XG5cbmNvbnN0IGNyZWF0ZURhdGEgPSAoXG4gIG1heERlZXA6IG51bWJlcixcbiAgbWF4Q2hpbGRyZW46IG51bWJlcixcbiAgbWluTm9kZXNOdW1iZXI6IG51bWJlcixcbiAgZGVlcCA9IDEsXG4gIGtleSA9ICdub2RlJ1xuKTogVHJlZVYyW10gPT4ge1xuICBsZXQgaWQgPSAwXG4gIHJldHVybiBBcnJheS5mcm9tKHsgbGVuZ3RoOiBtaW5Ob2Rlc051bWJlciB9KVxuICAgIC5maWxsKGRlZXApXG4gICAgLm1hcCgoKSA9PiB7XG4gICAgICBjb25zdCBjaGlsZHJlbk51bWJlciA9XG4gICAgICAgIGRlZXAgPT09IG1heERlZXAgPyAwIDogTWF0aC5yb3VuZChNYXRoLnJhbmRvbSgpICogbWF4Q2hpbGRyZW4pXG4gICAgICBjb25zdCBub2RlS2V5ID0gZ2V0S2V5KGtleSwgKytpZClcbiAgICAgIHJldHVybiB7XG4gICAgICAgIGlkOiBub2RlS2V5LFxuICAgICAgICBsYWJlbDogbm9kZUtleSxcbiAgICAgICAgY2hpbGRyZW46IGNoaWxkcmVuTnVtYmVyXG4gICAgICAgICAgPyBjcmVhdGVEYXRhKG1heERlZXAsIG1heENoaWxkcmVuLCBjaGlsZHJlbk51bWJlciwgZGVlcCArIDEsIG5vZGVLZXkpXG4gICAgICAgICAgOiB1bmRlZmluZWQsXG4gICAgICB9XG4gICAgfSlcbn1cblxuY29uc3QgcHJvcHN2MiA9IHtcbiAgdmFsdWU6ICdpZCcsXG4gIGxhYmVsOiAnbGFiZWwnLFxuICBjaGlsZHJlbjogJ2NoaWxkcmVuJyxcbn1cbmNvbnN0IGRhdGF2MiA9IGNyZWF0ZURhdGEoNCwgMzAsIDQwKVxuPC9zY3JpcHQ+XG5cbjxzdHlsZT5cbi5jdXN0b20tdHJlZS1ub2RlLWNvbnRhaW5lcntcblxufVxuLmlzLXBlbnVsdGltYXRlID4gLmVsLXRyZWUtbm9kZV9fY29udGVudCB7XG4gIGNvbG9yOiAjNjI2YWVmO1xufVxuXG4uZWwtdHJlZS1ub2RlLmlzLXBlbnVsdGltYXRlID4gLmVsLXRyZWUtbm9kZV9fY2hpbGRyZW4ge1xuICBkaXNwbGF5OiBmbGV4O1xuICBmbGV4LWRpcmVjdGlvbjogcm93O1xufVxuLmlzLXBlbnVsdGltYXRlID4gLmVsLXRyZWUtbm9kZV9fY2hpbGRyZW4gPiBkaXYge1xuICB3aWR0aDogMjUlO1xufVxuPC9zdHlsZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

When trying to style some -el-tree-node from tree-v2 we cant set a custom class

### What is Expected?

Have a tree v2 node with custom class

### What is actually happening?

Custom class can not be applied

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by zzjiaxiang**
close #16780

about #3912

### CODE REVIEW

File: packages/components/tree-v2/src/tree-node.vue [hunk 1]
```diff
@@ -79,27 +82,30 @@ const emit = defineEmits(treeNodeEmits)
...
-const handleDrop = (e: DragEvent) => {
-  emit('drop', props.node, e)
+  return isString(className) ? { [className]: true } : className
 }
-const handleExpandIconClick = () => {
-  emit('toggle', props.node)
-}
-const handleCheckChange = (value: CheckboxValueType) => {
+
+const handleClick = (e: MouseEvent) => emit('click', props.node, e)
```
Comment (btea) [inline packages/components/tree-v2/src/tree-node.vue:103]: For a function like this that does not directly return a value, I don't think it's necessary to refactor the abbreviation. I believe the original structure looks clearer.

File: docs/en-US/component/tree-v2.md [hunk 2]
```diff
@@ -57,6 +57,16 @@ tree-v2/custom-node
 
 :::
 
+## Custom node class
```
Comment (btea) [inline docs/en-US/component/tree-v2.md:60]:
```suggestion
## Custom node class ^(2.9.0)
```

**Review by btea — APPROVED** [review]
LGTM

**Review by warmthsea — APPROVED** [review]
Thanks!

**Pull request merged**